### PR TITLE
chore(package): update @types and TS

### PR DESCRIPTION
**Pull request by layershifter**
Extracted from #1042.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -128,7 +128,7 @@
     "tsconfig-paths": "^3.7.0",
     "tslint": "^5.11.0",
     "tslint-config-airbnb": "^5.11.1",
-    "typescript": "~3.2.2",
+    "typescript": "^3.3.3333",
```
Comment (bmdalex) [inline package.json:131]: yaaay 👍

File: packages/react/src/components/Flex/FlexItem.tsx [hunk 2]
```diff
@@ -80,9 +80,8 @@ class FlexItem extends UIComponent<Extendable<FlexItemProps>> {
       })
     }
 
-    return _.isNil(children)
-      ? children
-      : applyStyles(React.Children.only(children), styles, classes)
+    if (_.isNil(children)) return children
```
Comment (bmdalex) [inline packages/react/src/components/Flex/FlexItem.tsx:83]: we cannot use ternary operator anymore or was it your preference here?

**Review by bmdalex — APPROVED** [review]
looks good, hope it's safe to make this change at this time though 👍

File: packages/react/src/components/Flex/FlexItem.tsx (hunk 2 above)
Comment (layershifter) [reply packages/react/src/components/Flex/FlexItem.tsx:83]: It becomes unreadable:

```js
return _.isNil(children)
  ? children
  : applyStyles(
    React.Children.only(children) as React.ReactElement<any>,
    styles, 
    classes,
  )
```

**Pull request merged**